testing/format_trace/src/main.rs

```diff
@@ -21,3 +21,97 @@ fn main() {
 
     save_to_file(dest_filename, final_pretty_json);
 }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    fn generate_pretty_json(input_json: &str) -> String {
+        let ser_json = serde_json::from_str(&input_json).expect("Failed to parse the json input");
+        let prettified_json: String = prettify::prettify_value(ser_json, "", false);
+        let mut final_pretty_json: String = correct_path(&prettified_json);
+        final_pretty_json.push('\n'); //this is done automatically when saving the json to a file
+        final_pretty_json
+    }
+
+    #[test]
+    fn test_single_json_object() {
+        let input_json = r#"[{"Key":"val"}]"#;
+        let expected = r#"[
+  { "Key": "val" }
+]
+"#;
+        let final_pretty_json = generate_pretty_json(&input_json);
+        assert_eq!(final_pretty_json, expected);
+    }
+
+    #[test]
+    fn test_non_absolute_path_json() {
+        let input_json = r#"[{"Path":"?"},{"Path":"src/dir/main.nr"}]"#;
+        let expected = r#"[
+  { "Path": "?" },
+  { "Path": "src/dir/main.nr" }
```

Comment:
Interesting. So if there's nothing before the src, <relative-to-this> is not added.

Good to know! This is why tests are useful :)